mount/temp.go

```diff
@@ -0,0 +1,118 @@
+package mount
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/containerd/containerd/log"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	t, err := TempLocation("/tmp")
+	if err != nil {
+		panic(err)
+	}
+	DefaultTempLocation = t
+}
+
+var DefaultTempLocation TempMounts
+
+func TempLocation(root string) (TempMounts, error) {
+	root, err := filepath.Abs(root)
+	if err != nil {
+		return DefaultTempLocation, err
+	}
+	if err := os.MkdirAll(root, 0700); err != nil {
+		return DefaultTempLocation, err
+	}
+	return TempMounts{
+		root: root,
+	}, nil
+}
+
+type TempMounts struct {
+	root string
+}
+
+// WithTempMount mounts the provided mounts to a temp dir, and pass the temp dir to f.
+// The mounts are valid during the call to the f.
+// Finally we will unmount and remove the temp dir regardless of the result of f.
+func (t TempMounts) Mount(ctx context.Context, mounts []Mount, f func(root string) error) (err error) {
```

Comment:
The function name is confused. It look like that we don't do unmount in this function, but indeed we really have done it.

Reply:
Ya, I'm trying to figureout naming still and how you consume this from a package/type level.  Any ideas?

Reply:
Maybe TempMounts.Handle   and TempMounts.Cleanup :)